lib/src/data/persistence/cache/cache_policy.dart

```diff
@@ -0,0 +1,3 @@
+abstract class CachePolicy<T> {
+  bool isValid(T data);
```

Comment:
oprócz tego przydałaby się metoda w stylu `shouldFetch`, która by określała czy pobrać dane z api mimo ładowania z cache w tle

Reply:
mógłbyś szerzej to opisać? Obecnie chciałem, żeby isValid zwracało true, jeżeli najnowszy zacachowany element został opublikowany maksymalnie godzinę temu, no i jak jak jest starszy niż godzina, to zaciągnąć z api, dodać do bazy nowe i wyświetlić, jak jest świeży to wrzucić z bazy i nie dotykać api. Od czego zależałoby shouldFetch?

Reply:
`shouldFetch` mogłoby informować czy pobierać z api mimo, że wyciągamy z cache. Załóżmy, że artykuł nie jest starszy niż godzina i pobieramy z cache. Wtedy zamiast nie dotykać api moglibyśmy i tak w tle pobrać z api artykuł, żeby był świeży lub tego nie robić w zależności od tego co zwróci flaga `shouldFetch`. To podejście z jednego z sampli architektury androidych :p https://github.com/android/architecture-components-samples/blob/master/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.kt

Reply:
ok, kumam, pytanie tylko jaki dać warunek od którego zależałoby to shouldFetch? :)

Reply:
to tu właśnie będzie warunek typu czy starsze niż godzinę/dzień/tydzień lub po prostu true, w zależności od tego jakie założenie biznesowe

Reply:
tak myślę, że to isValid to dokładnie to robi, chodzi bardziej o to, że u Ciebie założyłeś, że jeżeli jest w cache to nigdy nie pobieramy z neta, a imo to pobranie z cache tego nie wyklucza

Reply:
założenie jest nieco inne, założenie jest takie, że dopóki najnowszy zacacheowany artykuł jest nie starszy niż maksymalnie godzina to bierzemy z cache, natomiast, jeżeli jest starszy to pobieramy z api, dorzucamy te świeższe do bazy i populujemy świeże dany z bazy jako source of truth do użytkownika, dlatego myślę jaki dać warunek dla tego shouldFetch bo true na sztywno to tak trochę mały zysk z cache'a :P

Reply:
zysk z cache taki, że user nie musi czekać na dane aż pobiorą się z api tylko dostaje je od razu lokalnie + w przypadku jak nie ma neta to dalej ma dane. Może pobierajmy dane z api zawsze pomimo brania ich z cache?

Reply:
czyli powiedzmy jeżeli isValid == true to pobieramy z api ale nie czekamy na wynik tylko propagujemy wynik z bazy, natomiast kiedy jest starszy niż godzina to pobieramy z api, czekamy na wynik i z bazy pobieramy świeże?

Reply:
dokładnie tak